chore(accounts): refactor server controllers

jniles opened the pull request:
This commit refactors the server account controllers to conform to 2.x API standards.  Here are the changes:
1. Improve documentation with JSDoc syntax.
2. Group account controllers in `finance/accounts/` directory.
3. Change account_type routes to `accounts/types`.

This should give a solid start to the #269.

---

Before submitting this pull request, please verify that you have:
- [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
- [x] Run integration tests.
- [x] Run end-to-end tests.
- [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

jniles committed:
chore(accounts): refactor server controllers

This commit refactors the server account controllers to conform to
2.x API standards.  Here are the changes:

 1. Improve documentation with JSDoc syntax.
 2. Group account controllers in `finance/accounts/` directory.
 3. Change account_type routes to `accounts/types`.

This should give a solid start to the #269.

sfount commented on server/controllers/finance/accounts/index.js:
```diff
@@ -0,0 +1,220 @@
+    `;
+  }
+
+  if (req.query.locked === '0') {
```
Does the implementation of this flag make sense from an API standpoint? 

For example: 

``` js
// returns only accounts that are not locked
'/accounts/:id?locked=0' 

// returns all accounts without considering locked property
'/accounts/:id?locked=1'
```

Either flags like this could implement both behaviours or it could be more of a binary flag 

For example:

``` js
// returns only accounts that are not locked
'/accounts/:id?activeAccounts=1'
```

What do you think?

jniles replied:
Unfortunately, the comment at the top is wrong: this API route is `/accounts` not, `accounts/:id`.

That aside, I've been thinking of the query strings as filters.  For example:

``` js
// this will return all records with no conditions
'/accounts'

// this will return all records with locked = 0
'/accounts?locked=0'

// this will return all records with locked = 1
'/accounts?locked=1'
```

This style mixes option flags (`detailed=1`) with filtering flags (`locked=1`).  If that is to confusing we can change this behavior.  What do you think?

sfount replied:
I have no problem with this! In this case I think it would be best to implement the `'?locked=1'` filter as well so that this case is not simple ignored.

sfount replied:
Also updating the comment would avoid future confusion!

jniles replied:
Will do.

jniles committed:
Fix(accounts): doc typo and implement ?locked=1

This commit fixes a type in the doc comments for the accounts controller
and implements filtering on `locked=1`.

sfount commented on server/controllers/finance/accounts/index.js:
```diff
@@ -97,11 +99,20 @@ function list(req, res, next) {
 
+  // if locked is a number, filter on it
+  if (!isNaN(locked)) {
+    sql += ` WHERE a.locked = ${locked}`;
```
This should do it! Looks good to me. Could you remove the debug logs?

jniles replied:
Whoops!  Took care of those.
